src/finite_elasticity_routines.f90

```diff
@@ -4008,6 +4031,14 @@ SUBROUTINE FINITE_ELASTICITY_GAUSS_CAUCHY_TENSOR(EQUATIONS_SET,DEPENDENT_INTERPO
           lambda_a=UP-(UP-LOW)/2.0_DP
 
           F_a_inv=0.0_DP
+          IF(lambda_a<TOL) THEN
+           LOCAL_ERROR="Warning: lambda_a is close to zero"
+           CALL FlagError(LOCAL_ERROR,ERR,ERROR,*999)
+!            WRITE(*,*) "UP: ", UP
```

Comment:
Great, thanks for the changes. Sure, I'll pull in. @klotz-t their may be a bug in the eigenvalue routines but it would be better to fix them as this would fix other code that needs eigenvalues. There is no problem using LAPACK but at the very least the LAPACK routine should be called from within the eigenvalue routine and not directly from where it is needed.